roles/cluster_deploy_prev_catalog_source/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Fail if catalog_image is undefined
+  fail:
+    msg: Must provide a catalog_image
+  when: catalog_image is undefined
+
+- name: Fail if catalog_registry is undefined
+  fail:
+    msg: Must provide a catalog_registry
+  when: catalog_registry is undefined
+
+- name: 'Deploy N-1 {{ catalog_image }} CatalogSource'
+  block:
+  - name: Set previous OCP version for the CatalogSource
+    set_fact:
+      catalog_version: "v{{ openshift_release_major }}.{{ openshift_release_minor|int - 1 }}"
```

Comment:
what about allowing `catalog_version` to be passed as a parameter? so that the role can be fully generic? and let `"v{{ openshift_release_major }}.{{ openshift_release_minor|int - 1 }}"` be the default value?

(mind though that Ansible is tricky on that, making the `block` conditioned by a `when: catalog_version | length == 0` may not be enough; that's where I use `fail; rescue;`)

Reply:
Good point. Ill try to do it a bit differently and avoid rescue blocks here. Let me know what you think once I push the changes.

Reply:
> Ill try to do it a bit differently and avoid rescue blocks

good luck with that 🙃 
make sure you triple test it, just in case :)
(eg, add a `fail` task right after setting one value or another and print the variable value)